### PR TITLE
Update typescript.md

### PR DESCRIPTION
**Pull request by JonghwanWon**
This PR updates typescript.md to Adds a few extra hints.

### CODE REVIEW

**Comment by ShivamJoker** [comment]
Good work

File: typescript.md [hunk 1]
```diff
@@ -204,10 +223,60 @@ export interface User { ... }
...
+
+type Str = First<['hello', 1, false]>; // 'hello'
+```
+
+## Literal Type
+
+```ts
+const point = { x: 4, y: 2 }; // { x: number, y: number }
+
+const literalPoint = point as const; // { x: 4, y: 2 };
```
Comment (lmarvaud) [inline typescript.md:271]: Hello,

I've the error :

```
const literalPoint = point as const;
                     -----
A 'const' assertions can only be applied to references to enum members, or string, number, boolean, array, or object literals.ts(1355)
```

Comment (JonghwanWon) [reply typescript.md]: Hello, @lmarvaud 

I fixed it, It was a incorrect example of the a 'const' assertion

File: typescript.md [hunk 2]
```diff
@@ -204,10 +223,60 @@ export interface User { ... }
...
+type GetReturnType<T> = T extends (...args: unknown[]) => infer R
+  ? R
+  : never;
+
+type Num = GetReturnType<() => number>; // number
+
+type First<T extends Array<any>> = T extends [infer F, ...infer Rest] ? F : never;
+
+type Str = First<['hello', 1, false]>; // 'hello'
+```
```
Comment (lmarvaud) [inline typescript.md:264]: those are two different examples. You may split those.

```ts
type GetReturnType<T> = T extends (...args: unknown[]) => infer R
  ? R
  : never;

type Num = GetReturnType<() => number>; // number
```

```ts
type First<T extends Array<any>> = T extends [infer F, ...infer Rest] ? F : never;

type Str = First<['hello', 1, false]>; // 'hello'
```

**Review by lmarvaud — APPROVED** [review]

File: typescript.md (hunk 2 above)
Comment (JonghwanWon) [reply typescript.md]: Thanks.
I fixed it.

**Pull request merged**